contracts/root/staking/CustomSupernetManager.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
+
+import "@openzeppelin/contracts/access/Ownable2Step.sol";
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./lib/SupernetManager.sol";
+import "../../interfaces/common/IBLS.sol";
+import "../../interfaces/IStateSender.sol";
+import "../../interfaces/root/staking/ICustomSupernetManager.sol";
+
+struct Validator {
+    uint256[4] blsKey;
+    uint256 stake;
+    bool isWhitelisted;
+    bool isActive;
+}
+
+contract CustomSupernetManager is ICustomSupernetManager, Ownable2Step, SupernetManager {
+    using SafeERC20 for IERC20;
+
+    bytes32 private constant STAKE_SIG = keccak256("STAKE");
+    bytes32 private constant UNSTAKE_SIG = keccak256("UNSTAKE");
+    bytes32 private constant SLASH_SIG = keccak256("SLASH");
+    uint256 private constant SLASHING_PERCENTAGE = 50;
+
+    IBLS private immutable BLS;
+    IStateSender private immutable STATE_SENDER;
+    IERC20 private immutable MATIC;
+    address private immutable CHILD_VALIDATOR_SET;
+    address private immutable EXIT_HELPER;
+
+    bytes32 public immutable DOMAIN;
+
+    mapping(address => Validator) public validators;
+
+    modifier onlyValidator(address validator) {
+        if (!validators[validator].isActive) revert Unauthorized("VALIDATOR");
+        _;
+    }
+
+    constructor(
+        address stakeManager,
+        address bls,
+        address stateSender,
+        address matic,
+        address childValidatorSet,
+        address exitHelper,
+        string memory domain
+    ) SupernetManager(stakeManager) {
+        BLS = IBLS(bls);
+        STATE_SENDER = IStateSender(stateSender);
+        MATIC = IERC20(matic);
+        CHILD_VALIDATOR_SET = childValidatorSet;
+        EXIT_HELPER = exitHelper;
+        DOMAIN = keccak256(abi.encodePacked(domain));
+    }
+
+    /**
+     * @inheritdoc ICustomSupernetManager
+     */
+    function whitelistValidators(address[] calldata validators_) external onlyOwner {
+        uint256 length = validators_.length;
+        for (uint256 i = 0; i < length; i++) {
+            _addToWhitelist(validators_[i]);
+        }
+    }
+
+    /**
+     * @inheritdoc ICustomSupernetManager
+     */
+    function register(address validator_, uint256[2] calldata signature, uint256[4] calldata pubkey) external {
+        Validator storage validator = validators[validator_];
+        if (!validator.isWhitelisted) revert Unauthorized("WHITELIST");
+        _verifyValidatorRegistration(validator_, signature, pubkey);
+        validator.blsKey = pubkey;
+        validator.isActive = true;
+        _removeFromWhitelist(validator_);
+        emit ValidatorRegistered(msg.sender, pubkey);
+    }
+
+    /**
+     * @inheritdoc ICustomSupernetManager
+     */
+    function withdrawSlashedStake(address to) external onlyOwner {
+        uint256 balance = MATIC.balanceOf(address(this));
```

Comment:
Slashed stake should not be withdrawable in this manner.

Reply:
what alternative do you suggest?

Reply:
Should remain in the contract imho, as future rewards.

Reply:
The Supernet should be able to decide what to do with the funds, could be paid out as rewards in the future or used to pay out any victims of attacks for example. So I'd say that the withdraw function is good for now unless we get any other specs